### PR TITLE
Convert E2E to XUnit 2.3.1

### PR DESCRIPTION
**Pull request by robward-ms**
### Issues

This pull request makes progress on issues #975, #939, #772, #628, #229, 

### Description

Convert E2E to XUnit 2.3.1
Convert E2E projects to NetFx 4.5.2
Consolidate E2E projects into a single project
Enable Xunit Analyzers & fix reported issues, including:
 - Assert.Equal(0, collection.Count) => Assert.Empty(collection)
 - Assert.Equal(1, collection.Count) => Assert.Single(collection)
 - Assert.Equal(var, const) => Assert.Equal(const, var)
 - Assert.True(collection.Contains(find)) => Assert.Contains(find, collection)
 - Remove duplicate Theory test cases
 - Cleanup unused Theory param warnings
 - Remove Assert.NotNull() on structs
Avoid .Wait() and .Results() in favor of async Task return for async unit tests
Cleanup unused test code.

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [x ] *Build and test with one-click build and test script passed*

### Additional work necessary

Port Unit Tests to NetCore
Port E2E Tests to NetCore

### CODE REVIEW

File: src/Microsoft.AspNet.OData/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -7,4 +7,4 @@
 [assembly: AssemblyTitle("Microsoft.AspNet.OData")]
 [assembly: AssemblyDescription("")]
```
Comment (xuzhg) [inline src/Microsoft.AspNet.OData/Properties/AssemblyInfo.cs]: remove unchanged file

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Aggregation/AggregationController.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        [EnableQuery]
+        public SingleResult<Customer> Get(int key)
+        {
+            ResetDataSource();
+            var db = new AggregationContext();
+            return SingleResult.Create(db.Customers.Where(c => c.Id == key));
+        }
+    }
+
+
```
Comment (xuzhg) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Aggregation/AggregationController.cs]: remove unused empty line

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Aggregation/AggregationEdmModel.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            var stdDevMethods = getCustomMethods(typeof(DbFunctions), StdDevMethodLabel);
+            CustomAggregateMethodAnnotation customMethods = new CustomAggregateMethodAnnotation();
+            customMethods.AddMethod(StdDevMethodToken, stdDevMethods);
+
+            model.SetAnnotationValue(model, customMethods);
+
+            return model;
+        }
+
+        private static Dictionary<Type, MethodInfo> getCustomMethods(Type customMethodsContainer, string methodName)
```
Comment (xuzhg) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Aggregation/AggregationEdmModel.cs]: >getCustomMethods [](start = 52, length = 16)

Rename the method name as "GetCustomMethods"

**Comment by AlanWong-MS** [comment]
We should track at a high level which groups of files are simply moved and not part of actual code churn.

**Pull request merged**